src/code42cli/cmds/search/options.py

```diff
@@ -127,6 +116,16 @@ def handle_parse_result(self, ctx, opts, args):
         return super().handle_parse_result(ctx, opts, args)
 
 
+def _convert_advanced_query_dict_to_filter_groups(ctx, param, arg):
+    if arg is None:
+        return
+    try:
+        filter_groups = [FilterGroup.from_dict(group) for group in arg["groups"]]
+        return filter_groups
+    except KeyError as e:
+        raise click.ClickException("Failed to validate query from JSON. {}".format(e))
```

Comment:
`parse` would probably be a better description of what's happening here than `validate`.

Reply:
Well, we've already parsed the json at this point, now we're trying to build the query, so I wanted it to be clear that it's not a json error but that we just couldn't build the query from the data we got. 

Should I just say "failed to build query"?

Reply:
Something like `failed to build query from input JSON` seems to fit that, yeah.